src/tiledimage.js

```diff
@@ -1148,27 +1150,44 @@ function compareTiles( previousBest, tile ) {
     return previousBest;
 }
 
-function drawTiles( tiledImage, lastDrawn ){
+function drawTiles( tiledImage, lastDrawn ) {
     var i,
         tile,
         tileKey,
         viewer,
         viewport,
         position,
-        tileSource;
+        tileSource,
+        needsRestore = false;
 
-    var usedClip = false;
-    if (tiledImage._clip) {
-        tiledImage._drawer.saveContext();
-        var box = tiledImage.imageToViewportRectangle(tiledImage._clip, true);
+    // TODO: Move this function
+    var boxToDrawerRectangle = function( box ) {
         var topLeft = tiledImage.viewport.pixelFromPoint(box.getTopLeft(), true);
         var size = tiledImage.viewport.deltaPixelsFromPoints(box.getSize(), true);
-        box = new OpenSeadragon.Rect(topLeft.x * $.pixelDensityRatio,
+
+        return new $.Rect(
+            topLeft.x * $.pixelDensityRatio,
             topLeft.y * $.pixelDensityRatio,
-            size.x * $.pixelDensityRatio,
-            size.y * $.pixelDensityRatio);
-        tiledImage._drawer.setClip(box);
-        usedClip = true;
+            size.x    * $.pixelDensityRatio,
+            size.y    * $.pixelDensityRatio
+        );
+    };
+
+    if ( tiledImage._clip ) {
+        tiledImage._drawer.saveContext();
+        var box = tiledImage.imageToViewportRectangle(tiledImage._clip, true);
+        var clipRect = boxToDrawerRectangle(box);
+
+        tiledImage._drawer.setClip(clipRect);
+        needsRestore = true;
+    }
+
+    if ( tiledImage.placeholderFillStyle && lastDrawn.length === 0 ) {
+        tiledImage._drawer.saveContext();
```

Comment:
If there is both _clip and placeholderFillStyle, this line will cause us to double save even though we only restore once. We don't need to save here if needsRestore is already set.